src/main/java/com/aws/greengrass/mqttbridge/MessageBridge.java

```diff
@@ -77,57 +76,62 @@ public void removeMessageClient(TopicMapping.TopicType clientType) {
     }
 
     private void handleMessage(TopicMapping.TopicType sourceType, Message message) {
-        String sourceTopic = message.getTopic();
-        LOGGER.atDebug().kv(LOG_KEY_SOURCE_TYPE, sourceType).kv(LOG_KEY_SOURCE_TOPIC, sourceTopic)
+        String fullSourceTopic = message.getTopic();
+        LOGGER.atDebug().kv(LOG_KEY_SOURCE_TYPE, sourceType).kv(LOG_KEY_SOURCE_TOPIC, fullSourceTopic)
                 .log("Message received");
 
         MessageClient sourceClient = messageClientMap.get(sourceType);
         if (sourceClient == null) {
-            LOGGER.atError().kv(LOG_KEY_SOURCE_TYPE, sourceType).kv(LOG_KEY_SOURCE_TOPIC, sourceTopic)
+            LOGGER.atError().kv(LOG_KEY_SOURCE_TYPE, sourceType).kv(LOG_KEY_SOURCE_TOPIC, fullSourceTopic)
                     .log("Source client not found");
             return;
         }
 
-        Map<String, List<Pair<String, TopicMapping.TopicType>>> srcDestMapping =
-                perClientSourceDestinationMap.get(sourceType);
+        Map<String, List<TopicMapping.MappingEntry>> srcDestMapping = perClientSourceDestinationMap.get(sourceType);
 
         if (srcDestMapping != null) {
-            final Consumer<Pair<String, TopicMapping.TopicType>> processDestination = destination -> {
-                MessageClient client = messageClientMap.get(destination.getRight());
+            final Consumer<TopicMapping.MappingEntry> processDestination = mapping -> {
+                MessageClient client = messageClientMap.get(mapping.getTarget());
                 // If the mapped topic string is empty string, we forward the message to the same topic as the
                 // source topic.
-                final String targetTopic = Utils.isEmpty(destination.getLeft()) ? sourceTopic : destination.getLeft();
+                final String baseTargetTopic = Utils.isEmpty(mapping.getTargetTopic())
+                    ? fullSourceTopic
+                    : mapping.getTargetTopic();
+                final String targetTopic = Utils.isEmpty(mapping.getTargetTopicPrefix())
+                    ? baseTargetTopic
+                    : String.join("/", mapping.getTargetTopicPrefix(), baseTargetTopic);
```

Comment:
I'm debating if we want to insert a topic level separator if the target topic prefix ends with one.Technically empty topic levels _are_ allowed (`prefix/topic//source/topic`) but I could see this easily tripping people up. Thoughts?

Reply:
Yea, I guess I shouldn't have assumed a path separator. What if we just directly concatenate the prefix to topic and defer any separator-decision to the person configuring the component.

Reply:
I like that better. It's flexible and easy to document. Thanks!